logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,38 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+                  Setting.new("node.name", String, Socket.gethostname),
+           BooleanSetting.new("config.allow_env", false),
+  ExistingFilePathSetting.new("config.path", nil, false),
+                  Setting.new("config.string", String, nil, false),
+           BooleanSetting.new("config.test", false),
+           BooleanSetting.new("config.auto_reload", false),
+                  Setting.new("config.reload_interval", Numeric, 3),
+           BooleanSetting.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
```

Comment:
This might be a comment of style, but I find it a bit weird to do a `.new() { }`, I havent seen that before in the context of  initialization. But I do agree it makes the code shorter and its valid ruby code. :)